### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by mathvbarone**

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -1,205 +0,0 @@
...
-  // UI DECLARATION
-  ui: {
-    form: document.querySelector(".form"),
-    fields: document.querySelectorAll(".input-field"),
-    inputs: {
-      name: document.querySelector(".name"),
-      email: document.querySelector(".email"),
-      message: document.querySelector(".message")
-    },
-    button: document.querySelector(".button"),
```
Comment (mblodorn) [inline js/script.js:11]: this name is not clear enough, what is this button for?

File: js/script.js [hunk 2]
```diff
@@ -1,205 +0,0 @@
-const startForm = {
-  // UI DECLARATION
-  ui: {
-    form: document.querySelector(".form"),
-    fields: document.querySelectorAll(".input-field"),
-    inputs: {
```
Comment (mblodorn) [inline js/script.js:6]: check all the inputs, if you can have just one of each, why use classes instead of ids?

File: js/script.js [hunk 3]
```diff
@@ -1,205 +0,0 @@
...
-      email: document.querySelector(".email"),
-      message: document.querySelector(".message")
-    },
-    button: document.querySelector(".button"),
-  },
-
-  // FUNCTIONS
-  functions: {
-    formValidation: () => {
-      const name = startForm.ui.inputs.name;
```
Comment (mblodorn) [inline js/script.js:17]: you could use destructors here to avoid repeating code, like:
```
const {email, message} = startForm.ui.inputs;
```

File: js/script.js [hunk 4]
```diff
@@ -1,205 +0,0 @@
...
-      const email = startForm.ui.inputs.email;
-      const message = startForm.ui.inputs.message;
-      const input = startForm.ui.fields;
-      const button = startForm.ui.button;
-      let erros = 0;
```
Comment (mblodorn) [inline js/script.js:22]: If you are not reassigning this variable it should be declared as a const, and the name should be "errors" instead of "erros"

File: js/script.js [hunk 5]
```diff
@@ -1,205 +0,0 @@
...
-          input.nextElementSibling.classList.remove("is-hidden");
-          erros++;
-        }
-      };
-
-      regexValidation(nameRegex, name);
-      regexValidation(emailRegex, email);
-      regexValidation(msgRegex, message);
-
-      erros === 0 ? (button.disabled = false) : (button.disabled = true);
```
Comment (mblodorn) [inline js/script.js:43]: you could shorten this sintax with something like
```
button.disabled = !!errors
```

And also remember that 0  == false in javascript

File: js/script.js [hunk 6]
```diff
@@ -1,205 +0,0 @@
...
-    },
-
-    formMessage: status => {
```
Comment (mblodorn) [inline js/script.js:46]: if and else is a better option in this block of code, since they belong to the same logic

File: js/script.js [hunk 7]
```diff
@@ -1,205 +0,0 @@
...
-    },
-
-    updateMsg: (nodes, msg) => {
-      nodes.insertAdjacentHTML("beforeend", msg);
-    },
-
-    return: () => {
-
-      const form = document.querySelector(".form");
-      const fields = document.querySelectorAll(".input-field");
```
Comment (mblodorn) [inline js/script.js:105]: why are you assigning the fields to another variable if you already have them in the ui variable?

File: js/script.js [hunk 8]
```diff
@@ -1,205 +0,0 @@
...
-
-      messageBox.forEach(box =>{
-        box.remove();
-      });
-    },
-
-    sendData: e => {
-      e.preventDefault();
-
-      const initFunctions = startForm.functions;
```
Comment (mblodorn) [inline js/script.js:122]: if you gonna name this initFunctions then you should find a better name for the startForm property.
Maybe startForm.initFunctions, so you can destruct it here (and no one will get confused):
```
const {initFunctions} = startForm
```

File: js/script.js [hunk 9]
```diff
@@ -1,205 +0,0 @@
...
-        if (request.readyState === 3) {
-          form.style.display = "none";
-          initFunctions.updateMsg(
-            container,
-            initFunctions.formMessage("loading")
-          );
-
-        }
-
-        else if (request.readyState === 4) {
```
Comment (mblodorn) [inline js/script.js:149]: please indent your code

File: js/script.js [hunk 10]
```diff
@@ -1,205 +0,0 @@
...
-
-          if (request.status == 200 && request.status < 300) {
-
-            setTimeout(function() {
```
Comment (mblodorn) [inline js/script.js:153]: why is this here?

File: js/script.js (hunk 2 above)
Comment (mathvbarone) [reply js/script.js:6]: Can you explain your point?

Whats the matter with classes instead of id's?

Comment (mblodorn) [reply js/script.js:6]: classes are meant to be reusable and ids are not. You may have troubles with your logic if you use these classes somewhere else in the future
